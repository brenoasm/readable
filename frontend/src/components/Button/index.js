import React from 'react';
import styled from 'styled-components';

import { colors } from '../../theme';

const defaultProps = {
  text: 'Button',
  textColor: colors.white,
  color: colors.primary.primaryOne,
  handleClick: () => {},
  type: 'button'
}

const StyledButton = styled.button`
  background-color: ${props => props.color};
  color: ${props => props.textColor};
  border-radius: 5px;
  font-weight: 600;
  font-size: 18px;
  cursor: pointer;
  padding: 5px;
`;

const onClick = (event, fn) => {
  event.stopPropagation();

  fn();
};

const Button = ({ type, textColor, color, handleClick, text }) => (
  <StyledButton
    type={type}
    onClick={(event) => onClick(event, handleClick)}
    textColor={textColor}
    color={color}>
    {text}
  </StyledButton>
);

Button.defaultProps = defaultProps;

export default Button;
