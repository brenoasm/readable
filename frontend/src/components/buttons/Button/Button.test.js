/* eslint-disable */

import React from 'react';

import Button from './';

describe('Button', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Button />);
  })

  afterEach(() => {
    wrapper.unmount();
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});
