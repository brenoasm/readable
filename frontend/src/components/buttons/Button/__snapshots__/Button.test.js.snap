// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    color="#236A62"
    disabled={false}
    handleClick={[Function]}
    text="Button"
    textColor="#fff"
    type="button"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Button",
      "color": "#236A62",
      "disabled": false,
      "onClick": [Function],
      "textColor": "#fff",
      "type": "button",
    },
    "ref": null,
    "rendered": "Button",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Button",
        "color": "#236A62",
        "disabled": false,
        "onClick": [Function],
        "textColor": "#fff",
        "type": "button",
      },
      "ref": null,
      "rendered": "Button",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
